src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/Snapshots/ReferenceItem.cs

```diff
@@ -1,18 +1,23 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Collections.Generic;
+
 using NuGet.SolutionRestoreManager;
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.NuGet
 {
+    /// <summary>
+    ///     Represents a single package, tool or project reference.
+    /// </summary>
     internal class ReferenceItem : IVsReferenceItem
     {
-        public ReferenceItem(string name, IVsReferenceProperties properties)
+        public ReferenceItem(string name, IEnumerable<IVsReferenceProperty> properties)
         {
             Requires.NotNullOrEmpty(name, nameof(name));
             Requires.NotNull(properties, nameof(properties));
 
             Name = name;
-            Properties = properties;
+            Properties = new ReferenceProperties(properties);
```

Comment:
Why introduce this logic here? It means the type is no longer a simple 'record'. Can `RestoreBuilder`'s methods just return `IVsReferenceProperties` and `IVsReferenceItems`? Does this make testing in your subsequent PR easier?

Reply:
It simplifies construction/merging in a later PR. It maps very strongly to other parallel concepts; you take an IEnumerable<T> and map to strongly-typed collections (such as Collection<T>/ReadOnlyCollection<T>).

Reply:
I've reverted this back, both you @Pilchie had similar feedback.